### PR TITLE
Generalize e2e pipeline test to work with keras and tensorflow saved models

### PR DESCRIPTION
**Pull request by OliverScherf**
The existing test pipeline already worked with both keras and tensorflow saved models, hence I generalized the naming and the content of the scripts. Since I renamed the script, I also updated the github build-and-test.yaml. 

Furthermore the test case generator can now handle int32 as output type of operations.

### CODE REVIEW

File: scripts/model_to_savedmodel_with_predict_function.py [hunk 1]
```diff
@@ -41,24 +41,32 @@ def translate(model_path: str, output_path: str, batch_size: int):
...
 
     tf.saved_model.save(module, output_path)
 
 
 def main():
     parser = argparse.ArgumentParser(
-        description="Translate keras model to saved model format")
+        description=
+        "Translates a model to saved model format with a predict function for further compiling." \
+        "Both keras and saved model format as input is supported."
```
Comment (marbre) [inline scripts/model_to_savedmodel_with_predict_function.py:56]:
```suggestion
       Keras and saved model format are supported as input.
```

**Review by marbre — CHANGES_REQUESTED** [review]
Minor nit.

File: scripts/generate_testscases.py [hunk 2]
```diff
@@ -120,15 +122,25 @@ def function_name():
                 )
 
             for i, curr_output in enumerate(outputs):
+                if not isinstance(curr_output, np.ndarray):
+                    if isinstance(curr_output, int):
```
Comment (simon-camp) [inline scripts/generate_testscases.py:126]: We should check if tf.int_64 is also represented as Python int here. It's worth adding an e2e test for both cases in a follow up.

**Review by simon-camp — APPROVED** [review]
LGTM. I'm not sure about the code reformatting though. We may want to a linter configuration file.

**Review by marbre — APPROVED** [review]

**Pull request merged**